### PR TITLE
Fix NodeJsWebAppFixtureIntegrationTest

### PR DESCRIPTION
**Pull request by bostko**
The pull request includes a few fixes

- it fixes the installation for npm on CentOS 7 ( @neykov I've tested it on the latest ubuntu and it works well)
- it fixes a general issue in the NodeJsWebAppFixtureIntegrationTest caused by an error in the way the nodejs entity is being stopped
- java version check which I hit in some other tests

These three things are in separate commits

### CODE REVIEW

File: software/webapp/src/test/java/brooklyn/entity/webapp/nodejs/NodeJsWebAppFixtureIntegrationTest.java [hunk 1]
```diff
@@ -143,7 +130,7 @@ public void testReportsServiceDownWhenKilled() throws Exception {
         LOG.info("test=testReportsServiceDownWithKilled; entity="+entity+"; app="+entity.getApplication());
         
         Entities.start(entity.getApplication(), ImmutableList.of(loc));
-        EntityTestUtils.assertAttributeEqualsEventually(MutableMap.of("timeout", 120*1000), entity, Startable.SERVICE_UP, true);
+        EntityTestUtils.assertAttributeEqualsEventually(MutableMap.of("timeout", 120 * 1000), entity, Startable.SERVICE_UP, true);
```
Comment (ahgittin) [inline software/webapp/src/test/java/brooklyn/entity/webapp/nodejs/NodeJsWebAppFixtureIntegrationTest.java]: can use `Duration.minutes(2)`

File: software/base/src/main/java/brooklyn/entity/java/JavaSoftwareProcessSshDriver.java [hunk 2]
```diff
@@ -336,7 +336,7 @@ protected int tryJavaInstall(String version, String command) {
         log.debug("Checking Java version at {}@{}", getEntity(), getLocation());
         // sed gets stdin like 'java version "1.7.0_45"'
         ProcessTaskWrapper<Integer> versionCommand = Entities.submit(getEntity(), SshTasks.newSshExecTaskFactory(
-                getLocation(), "java -version 2>&1 | grep \"java version\" | sed 's/.*\"\\(.*\\).*\"/\\1/'"));
+                getLocation(), "java -version 2>&1 | grep \" version\" | sed 's/.*\"\\(.*\\).*\"/\\1/'"));
```
Comment (ahgittin) [inline software/base/src/main/java/brooklyn/entity/java/JavaSoftwareProcessSshDriver.java]: worth a comment on the reason for this change -- presumably some java gives different output?

File: software/webapp/src/test/java/brooklyn/entity/webapp/nodejs/NodeJsWebAppFixtureIntegrationTest.java [hunk 3]
```diff
@@ -153,36 +140,16 @@ public void testReportsServiceDownWhenKilled() throws Exception {
...
-            newManagementContext.getRebindManager().rebind(getClass().getClassLoader(), null, ManagementNodeState.MASTER);
-            newManagementContext.getRebindManager().startPersistence();
-            SoftwareProcess entity2 = (SoftwareProcess) newManagementContext.getEntityManager().getEntity(tokill.getId());
-            entity2.stop();
-        } finally {
-            if (newManagementContext != null) ((ManagementContextInternal)newManagementContext).terminate();
-            Os.deleteRecursively(tempDir.getAbsolutePath());
-        }
-        LOG.info("called to stop {} in parallel mgmt universe", entity);
+        ((SoftwareProcessDriver)((DriverDependentEntity<?>) Entities.deproxy(tokill)).getDriver()).stop();
```
Comment (ahgittin) [inline software/webapp/src/test/java/brooklyn/entity/webapp/nodejs/NodeJsWebAppFixtureIntegrationTest.java]: nice!  much cleaner.

**Comment by ahgittin** [comment]
good fixes.  a few very minor comments, then good to merge.

File: utils/common/src/main/java/brooklyn/util/ssh/BashCommands.java [hunk 4]
```diff
@@ -376,7 +376,8 @@ public static String installPackageOr(Map<?,?> flags, String packageDefaultName,
                     chainGroup(
                         "echo yum exists, doing update",
                         ok(sudo("yum check-update")),
-                        sudo(yumInstall))));
+                        ok(sudo("yum -y install epel-release")),
+                            sudo(yumInstall))));
```
Comment (neykov) [inline utils/common/src/main/java/brooklyn/util/ssh/BashCommands.java]: bad indentation

File: software/webapp/src/test/java/brooklyn/entity/webapp/nodejs/NodeJsWebAppFixtureIntegrationTest.java (hunk 3 above)
Comment (bostko) [inline software/webapp/src/test/java/brooklyn/entity/webapp/nodejs/NodeJsWebAppFixtureIntegrationTest.java]: It was thanks to Svet

File: software/base/src/main/java/brooklyn/entity/java/JavaSoftwareProcessSshDriver.java (hunk 2 above)
Comment (bostko) [inline software/base/src/main/java/brooklyn/entity/java/JavaSoftwareProcessSshDriver.java]: I've included a comment in the commit.
Here is what `java -version` returns on my side
```
openjdk version "1.8.0_45"
OpenJDK Runtime Environment (build 1.8.0_45-b14)
OpenJDK 64-Bit Server VM (build 25.45-b02, mixed mode)
```
However we could consider using `grep -E "(openjdk|java) version"`

Comment (ahgittin) [inline software/base/src/main/java/brooklyn/entity/java/JavaSoftwareProcessSshDriver.java]: not sure how universal `grep -E` is so i'd be against that.  can you add a mention to `openjdk version "1.8.0_45"` in the comment?  things get missed in the git history, especially if not on the first line (although they shouldn't -- apologies from me for not noticing it!)

**Comment by bostko** [comment]
Regarding java version check change.
`java -version` returns different output if openjdk is used
Example:
```
openjdk version "1.8.0_45"
OpenJDK Runtime Environment (build 1.8.0_45-b14)
OpenJDK 64-Bit Server VM (build 25.45-b02, mixed mode)
```

File: software/webapp/src/test/java/brooklyn/entity/webapp/nodejs/NodeJsWebAppFixtureIntegrationTest.java [hunk 5]
```diff
@@ -191,7 +159,7 @@ public void testInitialNamedDeployments() {
         
         Entities.start(entity.getApplication(), ImmutableList.of(loc));
 
-        Asserts.succeedsEventually(MutableMap.of("timeout", 60*1000), new Runnable() {
+        Asserts.succeedsEventually(MutableMap.of("timeout", Duration.minutes(2)), new Runnable() {
```
Comment (neykov) [inline software/webapp/src/test/java/brooklyn/entity/webapp/nodejs/NodeJsWebAppFixtureIntegrationTest.java]: Is the timeout changed on purpose to two minutes?

**Comment by neykov** [comment]
good to merge

**Comment by neykov** [comment]
@bostko Could you close this.

**Pull request merged**